src/Renderer.php

```diff
@@ -214,20 +302,58 @@ public function inlineJson($values)
      */
     public function processData($data)
     {
-        $data['themePrefix'] = $this->theme ? $this->theme . "::" : "";
+        $themePrefix = $this->theme ? $this->theme . "::" : "";
+
         if (isset($data['stripes'])) {
-            $data['stripes'] = array_map(function ($stripe) {
+            // filter out stripes that don't exist for the theme
+            $filteredStripes = array_filter(
+                $data['stripes'],
+                fn($s) => $this->engine->exists($themePrefix . 'stripes/' . $s['type'])
+            );
+
+            // get an array of indexes
+            $indexes = array_keys($filteredStripes);
+
+            // process each stripe when any defined `processStripe[type]` functions
+            $processedStripes = array_map(function ($stripe, $index) {
                 $hook = 'processStripe' . ucwords(preg_replace("/^stripe\-/", '', $stripe['type']));
-                return is_callable([$this, $hook])
-                    ? $this->$hook($stripe)
-                    : $stripe;
-            }, array_filter($data['stripes'], function ($stripe) use ($data) {
-                return $this->engine->exists($data['themePrefix'] . 'stripes/' . $stripe['type']);
-            }));
+                return is_callable([$this, $hook]) ? $this->$hook($stripe, $index) : $stripe;
+            }, $filteredStripes, $indexes);
```

Comment:
—mostly updating our old stuff not your new stuff, but:

```
// Process each stripe with any defined `processStripe[type]` functions
$processedStripes = array_map(function ($stripe, $index) {
    $stripeName = str_replace('stripe-', '', $stripe['type']);
    $stripeName = ucwords($stripeName, '-');
    $stripeName = str_replace('-', '', $stripeName);
    $hook = "processStripe{$stripeName}";
    return is_callable([$this, $hook]) ? $this->$hook($stripe, $index) : $stripe;
}, $filteredStripes, $indexes);
```

(`str_replace` instead of `preg_replace` since there's no regex; actually handle multi-word stripe types)